dashboard/app/controllers/scripts_controller.rb

```diff
@@ -33,6 +33,13 @@ def create
   end
 
   def destroy
+    # Though @script.name is restricted to containing lower case characters,
+    # dashes, and numbers only, we do this (security) check for safety.
+    if (@script.name.start_with? '../') ||
+      (@script.name.include? '/../')
```

Comment:
This is still vulnerable to [triple-dot traversal](https://cwe.mitre.org/data/definitions/32.html) - some shells permit `cd ...` to go up two levels, etc.  It would probably be better to block any script name with two or more consecutive dots:

```ruby
if /\.{2,}/ =~ @script.name
  raise ArgumentError #...
end
```